doc/man3/flux_content_load.adoc

```diff
@@ -0,0 +1,144 @@
+flux_content_load(3)
+====================
+:doctype: manpage
+
+
+NAME
+----
+flux_content_load, flux_content_load_get, flux_content_store, flux_content_store_get, flux_content_store_get_valref, flux_content_destroy - load/store content
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ struct flux_content;
+
+ struct flux_content *flux_content_load (flux_t *h,
+                                         const char *ref,
+                                         int flags);
+
+ int flux_content_load_get (struct flux_content *r,
+                            void *buf,
+                            size_t *len);
+
+
+ struct flux_content *flux_content_store (flux_t *h,
+                                         const void *buf,
+                                         size_t len,
+                                         size_t maxblob,
+                                         int flags);
+
+ int flux_content_store_get (struct flux_content *r,
+                             const char **ref);
+
+ int flux_content_store_get_valref (struct flux_content *r,
+                                    const char **ref);
+
+ void flux_content_destroy (struct flux_content *r);
+
+DESCRIPTION
+-----------
+
+The Flux content store is an append-only, immutable, content addressed
+data store unique to each Flux instance, described in RFC 10.
+
+`flux_content_load()` sends a request to the content service
+to load a data blob by _ref_ which must be a valid "blobref" or "valref"
+(see below).  A _flux_content_ object is returned.  _flags_ is a mask
+that may include the values described below.
+
+`flux_request_load_get()` completes a load operation, blocking on
+response(s) if needed, parsing the result, and returning the requested
+blob in _buf_ and its length in _len_.  _buf_ is valid until
+`flux_content_destroy()` is called.  This function is idempotent.
+
+`flux_content_store()` sends a request to the content service
+to store a data blob _buf_ of length _len_.  If _maxblob_ is nonzero,
+_buf_ may be split into multiple store requests internally.
+If the request is split, all requests except possibly the last
+will be of size _maxblob_.  _flags_ is a mask that may include the
+values described below.
+
+`flux_content_store_get()` completes a store operation, blocking on
+response(s) if needed, parsing the result, and returning a blobref that can
+be used later to retrieve the stored blob.  The blobref is valid until
+`flux_content_destroy()` is called.  If the store request was split
+into multiple store requests internally, this function will fail with errno
+set to EFBIG, and `flux_content_store_get_valref()` should be used to
+retrieve the result as a treeref instead instead.  This function is
+idempotent.
```

Comment:
I could be confused, but should "retrieve the result as treeref" here be "retrieve results as valref" instead? I'm basing this solely on the function name "`flux_content_store_get_valref`" as I don't quite understand the API fully yet.